ashpy/metrics/gan.py

```diff
@@ -13,43 +13,76 @@
 # limitations under the License.
 
 """GAN metrics."""
+from __future__ import annotations
+
 import os
 import types
+from typing import TYPE_CHECKING, Callable, List, Tuple
 
-import tensorflow as tf
-import tensorflow_hub as hub
+import tensorflow as tf  # pylint: disable=import-error
 
+import tensorflow_hub as hub  # pylint: disable=import-error
 from ashpy.metrics import ClassifierMetric, Metric
 from ashpy.modes import LogEvalMode
 
+if TYPE_CHECKING:
+    import numpy as np
+    from ashpy.contexts import (  # pylint: disable=ungrouped-imports
+        BaseContext,
+        ClassifierContext,
+        GANContext,
+        GANEncoderContext,
+    )
+
 
 class DiscriminatorLoss(Metric):
-    """The ash way of measuring the discriminator loss value."""
+    """The Discriminator loss value."""
 
     def __init__(
-        self, model_selection_operator=None, logdir=os.path.join(os.getcwd(), "log")
-    ):
+        self,
+        model_selection_operator: Callable = None,
+        logdir: str = os.path.join(os.getcwd(), "log"),
+    ) -> None:
         """
+        Initialize the Metric.
+
         Args:
-            model_selection_operator: The operation to be used when model_selection
-                is on to compare the metrics. E.g.: operator.gt.
-            logdir: The log dir in which to save logs.
-        """
+            model_selection_operator (:py:obj:`typing.Callable`): The operation that will
+                be used when `model_selection` is triggered to compare the metrics,
+                used by the `update_state`.
+                Any :py:obj:`typing.Callable` behaving like an :py:mod:`operator` is accepted.
+
+                .. note::
+                    Model selection is done ONLY if an operator is specified here.
 
+            logdir (str): Path to the log dir, defaults to a `log` folder in the current
+                directory.
+
+        """
         super().__init__(
             name="d_loss",
             metric=tf.metrics.Mean(name="d_loss", dtype=tf.float32),
             model_selection_operator=model_selection_operator,
             logdir=logdir,
         )
 
-    def result(self):
-        return self._metric.result().numpy()
+    # def result(self) -> np.ndarray:
```

Comment:
Do we need to keep this comments?

Reply:
We can delete these sections, they should be handled by the parent class.

Reply:
Solved by ccf5ed6